packages/rspack/src/fileSystem.ts

```diff
@@ -12,33 +13,48 @@ function createThreadsafeNodeFSFromRaw(
 	fs: typeof import("fs")
 ): ThreadsafeWritableNodeFS {
 	return {
-		writeFile: (file, data) => fs.writeFileSync(file, data),
-		removeFile: file => fs.unlinkSync(file),
-		mkdir: dir => fs.mkdirSync(dir),
-		mkdirp: dir =>
-			fs.mkdirSync(dir, {
+		writeFile: (file, data) => {
+			return util.promisify(fs.writeFile.bind(fs))(file, data);
```

Comment:
What about `node:fs/promises`?

Reply:
Users can pass self-defined `"fs"` instances, for example `"memfs"`.